vm/gc.go

```diff
@@ -0,0 +1,112 @@
+package vm
+
+import (
+	"time"
+
+	"github.com/jbenet/goprocess"
+	"github.com/libp2p/go-smart-record/ir"
+)
+
+func (v *vm) gcLoop(proc goprocess.Process) {
+	msgSyncTicker := time.NewTicker(v.gcPeriod)
```

Comment:
You want to schedule a new tick explicitly each time after garbage collection runs. This accounts accurately for the time spent garbage collecting (which could be hours if the user puts their computer to sleep).

The issue with using a periodic ticker is: What happens if one garbage collection takes more time the tick interval? Calls to gc start happening one after the other, because the ticker events keep piling.

Reply:
You are right! I missed that detail. Thanks!